sites/example-project/src/pages/api/status/[...route]/extractQueries.server.js

```diff
@@ -46,66 +46,58 @@ const updateDirectoriesandStatus = function (queries, routeHash) {
 export const getStatusAndExtractQueries = function (route) {
 	let routeHash = md5(route);
 	let fileRoute = `./src/pages/${route}/+page.md`;
-	let content = fs.readFileSync(fileRoute);
-	content = content ? content.toString() : null;
+	let content = fs.readFileSync(fileRoute, 'utf-8');
 
-	if (content) {
-		let partialInjectedContent = preprocessor.injectPartials(content.toString());
-		let queries = preprocessor.extractQueries(partialInjectedContent);
+	let partialInjectedContent = preprocessor.injectPartials(content);
+	let queries = preprocessor.extractQueries(partialInjectedContent);
 
-		// Handle query chaining:
-		let maxIterations = 15;
-		let queryIds = queries.map((d) => d.id);
+	// Handle query chaining:
+	let maxIterations = 15;
+	let queryIds = queries.map((d) => d.id);
 
-		for (let i = 0; i <= maxIterations; i++) {
-			queries.forEach((query) => {
-				let references = query.compiledQueryString.match(/\${.*?\}/gi);
-				if (references) {
-					query.compiled = true;
-					references.forEach((reference) => {
-						try {
-							let referencedQueryID = reference.replace('${', '').replace('}', '').trim();
-							if (!queryIds.includes(referencedQueryID)) {
-								let errorMessage =
-									'Compiler error: ' +
-									(referencedQueryID === ''
-										? 'missing query reference'
-										: "'" + referencedQueryID + "'" + ' is not a query on this page');
-								throw new Error(errorMessage);
-							} else if (i >= maxIterations) {
-								throw new Error(circularRefErrorMsg);
-							} else {
-								const referencedQuery = queries.filter((d) => d.id === referencedQueryID)[0];
-								if (!query.inline && referencedQuery.inline) {
-									throw new Error(
-										`Cannot reference inline query from SQL File. (Referenced ${referencedQueryID})`
-									);
-								}
-								const queryString = `(${referencedQuery.compiledQueryString})`;
-								query.compiledQueryString = query.compiledQueryString.replace(
-									reference,
-									queryString
+	for (let i = 0; i <= maxIterations; i++) {
```

Comment:
This has far too many levels of nesting; can we do some refactoring in this PR to make this more legible?

Also a good opportunity to write tests before refactoring to ensure there aren't regressions

Reply:
Is it worth refactoring and adding tests considering [it's all ripped out in next](https://github.com/evidence-dev/evidence/blob/next/sites/example-project/src/pages/api/status/%5B...route%5D/extractQueries.server.js)?